src/internal/user_api.rs

```diff
@@ -259,23 +259,34 @@ impl UserDevice {
 }
 
 /// Claims required to form a valid [Jwt](struct.Jwt.html).
+///
+/// These are a mixture of public claims (predefined by the JWT standard - sub, iat, exp)
+/// and private claims (pid, sid, kid, uid). The private claims can be protected from
+/// collision by prefixing the claim names with "http://ironcore/". Some JWT generators
+/// such as Auth0 require a namespace on private claims.
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub struct JwtClaims {
     /// Unique user ID
     pub sub: String,
-    /// Project ID
-    pub pid: usize,
-    /// Segment ID
-    pub sid: String,
-    /// Service key ID
-    pub kid: usize,
     /// Issued time (seconds)
     pub iat: u64,
     /// Expiration time (seconds)
     ///
     /// We recommend it be set to `iat + 120`. The IronCore server will not use the value,
     /// and it will automatically reject JWTs that are received more than 120 seconds past `iat`.
     pub exp: u64,
+    /// Project ID
+    #[serde(alias = "http://ironcore/pid")]
+    pub pid: u32,
+    /// Segment ID
+    #[serde(alias = "http://ironcore/sid")]
+    pub sid: String,
+    /// Service key ID
+    #[serde(alias = "http://ironcore/kid")]
+    pub kid: u32,
+    /// User ID
+    #[serde(alias = "http://ironcore/uid")]
+    pub uid: Option<String>,
```

Comment:
I haven't seen this claim before. Is it only used for Auth0? Can we get an explanation in the comments? From what we have now, I can't tell the difference between this and `sub`.

Reply:
Yes, I guess. Auth0 fills in `sub` with some GitHub ID, and it puts the user's email into the `uid` claim. `ironcore-id` currently overwrites `sub` with `uid` if it is present.

Reply:
This should go into the doc comments either on `uid` or the whole `JwtClaims` (or both)